2023/src/day5/day5main.rs

```diff
@@ -21,11 +23,62 @@ pub fn day5_main() {
     day5_part1(&amc);
   });
   println!("\n[d5p1] took: {:#?}", fun_time);
+
+  fun_time = run_and_time(|| {
+    day5_part2(&amc);
+  });
+  println!("\n[d5p2] took: {:#?}", fun_time);
+}
+
+fn day5_part2(amc: &Almanac) {
+  println!("\n~ Day 5 Part 2 ~");
+  /*
+    Try a brute force method.
+      [note] - worked, but took 21gig RAM & 2800 seconds
+    
+    More optimal:
+    Pass in seed ranges,
+      and return any subranges that are within bounds
+  */
   
+  let seed_pairs: Vec<AlmanacPair> = amc.seeds
+    .chunks(2)
+    .map(|seed_pair| {
+      AlmanacPair {
+        start: seed_pair[0],
+        end: seed_pair[1],
+      }
+    })
+    .collect();
+
+
+  let mut locatons = Vec::new();
+  for seed_pair in seed_pairs.iter() {
```

Comment:
If you remove the `iter()` here, it'll by default call `into_iter()` and take ownership of `seed_pairs`.  I think that way you can also remove the `clone()` on the next line, since you own each `seed_pair`.

In general, it's preferable to take ownership when you can, rather than dealing with references everywhere.  Won't make much difference here, I guess it's more of a code cleanliness thing.

Reply:
omg I've been using `.iter()` everywhere - this is so good to know, thanks!